src/Services/LightningService.cs

```diff
@@ -910,24 +902,34 @@ public async Task<(PSBT?, bool)> GenerateTemplatePSBT(ChannelOperationRequest? c
                     .SetChange(changeAddress.Address)
                     .SendEstimatedFees(feeRateResult.FeeRate);
 
-                result.Item1 = builder.BuildPSBT(false);
+                var originalPSBT = builder.BuildPSBT(false);
+
+                //Hack to remove outputs
+                var combinedPsbTtx = originalPSBT.GetGlobalTransaction();
+                if (channelOperationRequest.Changeless)
+                {
+                    combinedPsbTtx.Outputs.Clear();
+                }
 
+                result.Item1 = combinedPsbTtx.CreatePSBT(network);
+
+                //Hack to make sure that witness and non-witness UTXOs are added to the PSBT
                 //Hack, see https://github.com/MetacoSA/NBitcoin/issues/1112 for details
                 foreach (var input in result.Item1.Inputs)
                 {
-                    input.SighashType = SigHash.None;
+                    input.WitnessUtxo = originalPSBT.Inputs.FirstOrDefault(x=> x.PrevOut == input.PrevOut)?.WitnessUtxo;
+                    input.NonWitnessUtxo = originalPSBT.Inputs.FirstOrDefault(x=> x.PrevOut == input.PrevOut)?.NonWitnessUtxo;
```

Comment:
Sighash was deleted by mistake prob.
```suggestion
                    input.WitnessUtxo = originalPSBT.Inputs.FirstOrDefault(x=> x.PrevOut == input.PrevOut)?.WitnessUtxo;
                    input.NonWitnessUtxo = originalPSBT.Inputs.FirstOrDefault(x=> x.PrevOut == input.PrevOut)?.NonWitnessUtxo;
                    input.SighashType = SigHash.None;
```